jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt

```diff
@@ -50,10 +50,12 @@ class CodeWhispererRecommendationManager {
         }
         val rangeMarkers = mutableMapOf<RangeMarker, Reference>()
         recommendation.references().forEach {
+            val tempEnd = invocationStartOffset + it.recommendationContentSpan().end()
+            val referenceEnd = if (tempEnd <= endOffset) tempEnd else endOffset
             rangeMarkers[
                 tempDocument.createRangeMarker(
                     invocationStartOffset + it.recommendationContentSpan().start(),
-                    invocationStartOffset + it.recommendationContentSpan().end()
+                    referenceEnd
```

Comment:
This will give you an invalid rangeMarker error when the reference range is truncated entirely if the start of the reference is greater than endOffset, since we are creating the rangeMarker in the tempDocument, we are interested in the recommendation string that's being inserted to this document, which is already reformatted. Thus, in order to create rangeMarkers from this reformatted string, we have to remove all the ones that are outside of the truncated range. This means you have to check the following:
```
val referenceStart = invocationStartOffset + it.recommendationContentSpan().start()
if (referenceStart >= endOffset) return@forEach
```